File: Reversi/Util/Coords.cs
namespace Reversi.Util
{
    public struct Coords
    {
        public int X { get; set; }
        public int Y { get; set; }

        public Coords(int x, int y)
        {
            X = x;
            Y = y;
        }

        public override string ToString()
        {
            return $"({X}, {Y})";
        }

        public static Coords operator +(Coords first, Coords second) => new Coords()
        {
            X = first.X + second.X,
            Y = first.Y + second.Y
        };

        public static Coords operator -(Coords first, Coords second) => new Coords()
        {
            X = first.X - second.X,
            Y = first.Y - second.Y
        };

        public static bool operator ==(Coords first, Coords second) => first.X == second.X &&
                                                                       first.Y == second.Y;

        public static bool operator !=(Coords first, Coords second) => !(first == second);
    }
}

File: Reversi-Website/Static/src/js/modules/reversi/board/coords.js
/**
 *
 * @param {number} x
 * @param {number} y
 * @constructor
 */
Reversi.Coords = function (x, y) {
    /**
     *
     * @type {number}
     */
    this.x = x;

    /**
     *
     * @type {number}
     */
    this.y = y;

    /**
     * @description returns 'x, y'
     * @return {string}
     */
    this.toString = () => {
        return x + ", " + y
    };

    /**
     *
     * @param {Reversi.Coords} otherCoords
     */
    this.equals = function (otherCoords) {
        return this.x === otherCoords.x && this.y === otherCoords.y
    }
};
/**
 *
 * @param {string} input
 * @returns {Coords}
 */
Reversi.Coords.parse = (input) => {

    let input_split = input.split(",");

    let isCorrectLength = input_split.length === 2;

    let isCorrectType = true;

    input_split.map(item => {
        item = item.trim();

        if (isNaN(Number(item)))
            isCorrectType = false;
    });


    if (!isCorrectLength || !isCorrectType)
        throw new Error("Incorrect input: " + input);

    let x = Number(input_split[0]);
    let y = Number(input_split[1]);

    return new Reversi.Coords(x, y);
};